trunk/SUAVE/Methods/Noise/Fidelity_One/Propeller/compute_harmonic_noise.py

```diff
@@ -118,53 +120,43 @@ def compute_harmonic_noise(harmonics,freestream,angle_of_attack,position_vector,
-    p_mT_H            = abs(p_mT_H)             
-
-    # sound pressure for loading noise 
-    p_mL_H_function   = (m_1d*B*M_t*np.sin(theta_r)/ (2*np.sqrt(2)*np.pi*Y*R_tip)) *exponent_fraction
-    p_mL_H_integral   = np.trapz((((np.cos(theta_r_prime)/(1 - M_x*np.cos(theta_r)))*dT_dr - (1/((r**2)*M_t*R_tip))*dQ_dr)
-                                  * np.exp(1j*phi_s)*Jmb * psi_L),x = r[0,0,0,:,0], axis = 3 )
-    p_mL_H            =  p_mL_H_function[:,:,:,0]*p_mL_H_integral 
-    p_mL_H            =  abs(p_mL_H)  
-
-    # unweighted harmonic sound pressure level 
-    #res.SPL_r[i,:,p_idx,:]      = 20*np.log10((np.linalg.norm(p_mL_H + p_mT_H, axis = 0))/p_ref) 
-    res.SPL_r        = 20*np.log10((abs(p_mL_H + p_mT_H))/p_ref) 
-    res.p_pref_r     = 10**(res.SPL_r/10)   
-    res.SPL_r_dBA    = A_weighting(res.SPL_r,res.f[:,:,:,0,:]) 
-    res.p_pref_r_dBA = 10**(res.SPL_r_dBA/10)         
-        
-    # convert to 1/3 octave spectrum 
-    res.SPL_prop_h_spectrum     = SPL_harmonic_to_third_octave(res.SPL_r,res.f[:,0,0,0,:],settings)         
-    res.SPL_prop_h_dBA_spectrum = SPL_harmonic_to_third_octave(res.SPL_r_dBA,res.f[:,0,0,0,:],settings)     
-  
-    # Rotational(periodic/tonal)   
-    res.SPL_prop_tonal_spectrum = 10*np.log10( 10**(res.SPL_prop_h_spectrum/10)) 
- 
+    p_mT_H_integral   = -((M_r**2)*(t_c)*np.exp(1j*phi_s)*Jmb*(k_x**2)*psi_V ) * ((rho*(a**2)*B*np.sin(theta_r))/(4*np.sqrt(2)*np.pi*(Y/D)))* exponent_fraction
+    p_mT_H            = np.trapz(p_mT_H_integral,x = r[0,0,0,:,0], axis =3) 
+    
+    p_mT_H_abs        = abs(p_mT_H)             
+    p_mL_H_integral   = (((np.cos(theta_r_prime)/(1 - M_x*np.cos(theta_r)))*dT_dr - (1/((r**2)*M_t*R_tip))*dQ_dr)
+                                  * np.exp(1j*phi_s)*Jmb * psi_L)*(m_1d*B*M_t*np.sin(theta_r)/ (2*np.sqrt(2)*np.pi*Y*R_tip)) *exponent_fraction
+    p_mL_H            = np.trapz(p_mL_H_integral,x = r[0,0,0,:,0], axis = 3 ) 
+    p_mL_H_abs        =  abs(p_mL_H)  
+    
+    # sound pressure levels  
+    res.SPL_prop_harmonic_bpf_spectrum     = 20*np.log10((abs(p_mL_H_abs + p_mT_H_abs))/p_ref) 
+    res.SPL_prop_harmonic_bpf_spectrum_dBA = A_weighting(res.SPL_prop_harmonic_bpf_spectrum,res.f[:,:,:,0,:]) 
+    res.SPL_prop_harmonic_1_3_spectrum     = SPL_harmonic_to_third_octave(res.SPL_prop_harmonic_bpf_spectrum,res.f[:,0,0,0,:],settings)         
+    res.SPL_prop_harmonic_1_3_spectrum_dBA = SPL_harmonic_to_third_octave(res.SPL_prop_harmonic_bpf_spectrum_dBA,res.f[:,0,0,0,:],settings)  
+    res.SPL_prop_harmonic_1_3_spectrum[np.isinf(res.SPL_prop_harmonic_1_3_spectrum)] = 0
+    res.SPL_prop_harmonic_1_3_spectrum_dBA[np.isinf(res.SPL_prop_harmonic_1_3_spectrum_dBA)] = 0
+    
     return  
 
 ## @ingroupMethods-Noise-Fidelity_One-Propeller 
-def vectorize(X,num_cpt,num_h,num_r,num_prop,num_mic,vectorize_method): 
+def vectorize(X0,num_cpt,num_h,num_r,num_prop,num_mic,num_azi,vectorize_method): 
```

Comment:
again, issues here;
also you already have this defined in the compute_source_coordinates, so this looks to be repeated

Reply:
Removed the vectorize function